### PR TITLE
[7.x] Better ScriptHandler.php, migrate from drustack/drustack-standard

### PR DESCRIPTION
**Pull request by hswong3i**
Migrate from https://github.com/drustack/drustack-standard/blob/master/src/Composer/ScriptHandler.php, include:
- Original script handlers, clone from Drupal 8.x Drupal\Core\Composer\Composer
- Generate info meta data into .info, logic clone from drush 8.x

### CODE REVIEW

**Comment by hswong3i** [comment]
Some update due to https://www.drupal.org/node/2760467 changes.

File: scripts/composer/ScriptHandler.php [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+        $root = getcwd().'/web';
+
+        // Prepare the settings file for installation.
+        if ($fs->exists($root.'/sites/default/default.settings.php')
+            && !$fs->exists($root.'/sites/default/settings.php')) {
+            $fs->copy(
+                $root.'/sites/default/default.settings.php',
+                $root.'/sites/default/settings.php'
+            );
+            $fs->chmod($root.'/sites/default/settings.php', 0666);
```
Comment (madalinignisca) [inline scripts/composer/ScriptHandler.php:242]: Why?

File: scripts/composer/ScriptHandler.php [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+        }
+
+        // Prepare the services file for installation.
+        if ($fs->exists($root.'/sites/default/default.services.yml')
+            && !$fs->exists($root.'/sites/default/services.yml')) {
+            $fs->copy(
+                $root.'/sites/default/default.services.yml',
+                $root.'/sites/default/services.yml'
+            );
+            $fs->chmod($root.'/sites/default/services.yml', 0666);
```
Comment (madalinignisca) [inline scripts/composer/ScriptHandler.php:253]: Why?

File: scripts/composer/ScriptHandler.php [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+            $event->getIO()->write('Create a sites/default/services.yml file with chmod 0666');
+        }
+
+        // Create the files directory with chmod 0777.
+        if (!$fs->exists($root.'/sites/default/files')) {
+            $oldmask = umask(0);
+            $fs->mkdir($root.'/sites/default/files', 0777);
```
Comment (madalinignisca) [inline scripts/composer/ScriptHandler.php:260]: Really why???

**Review by madalinignisca — COMMENTED** [review]
Why should permissions be set so open? Permissions responsibility should be an important part before deploying live and not automated like this.

**Comment by jcnventura** [comment]
I've been syncing the 7.x branch with 8.x. Other than the vendor cleanup stuff, I'm not sure why all the code that is in drustack/drustack-standard but not in drupal-composer/drupal-project is needed. Then again, I don't see the reason to even cleanup anything.

**Pull request closed without merging**